pkg/user/handler.go

```diff
@@ -72,6 +72,27 @@ func (h Handler) SignUp(c *gin.Context) {
 	c.JSON(http.StatusCreated, user)
 }
 
+func handleSignUpErrors(err error) error {
+	validationErrors, ok := err.(validator.ValidationErrors)
+	if ok {
```

Comment:
I would invert the condition to reduce nesting. So `if !ok { return errdef.NewBadRequest("Error binding data: %+v", err) }` then the field error logic 😄